code/modules/spells/artifacts.dm

```diff
@@ -150,18 +150,550 @@
+	..()
+	if(isliving(A))
+		var/mob/living/L = A
+		L.apply_effect(2,INCINERATE)
+		L.electrocute_act(10, tesla_shock = 1)
+		if(L.find_type() & TYPE_WEIRD)
+			L.apply_effect(2,INCINERATE)
+			L.adjustFireLoss(5)
+
+/obj/item/weapon/sunlight_spear/throw_at()
+	..()
+	var/obj/item/weapon/sunlight_scroll/S
+	if(user.l_hand && istype(user.l_hand,/obj/item/weapon/sunlight_scroll))
+		S = user.l_hand
+	else if(user.r_hand && istype(user.r_hand,/obj/item/weapon/sunlight_scroll))
+		S = user.r_hand
+	else
+		S = new(get_turf(src))
+		user.put_in_hands(S)
+
+/obj/item/weapon/sunlight_spear/throw_impact(atom/hit_atom)
+	..()
+	explosion(hit_atom, -1, 0, 2)
+	playsound(user, 'sound/magic/lightningbolt.ogg', 60, 1)
+
+	for(var/obj/structure/closet/L in hear(1, get_turf(hit_atom)))
+		if(locate(/mob/living/carbon/, L))
+			for(var/mob/living/carbon/M in L)
+				bang(get_turf(src), M)
+
+	for(var/mob/living/carbon/M in hear(1, get_turf(hit_atom)))
+		bang(get_turf(src), M)
+
+	for(var/obj/effect/blob/B in hear(2,get_turf(hit_atom)))       		//Blob damage here
+		var/damage = round(30/(get_dist(B,get_turf(hit_atom))+1))
+		B.health -= damage
+		B.update_icon()
+
+	if(isliving(hit_atom))
+		var/mob/living/L = hit_atom
+		L.apply_effect(4,INCINERATE)
+		L.electrocute_act(20, tesla_shock = 1)
+		if(L.find_type() & TYPE_WEIRD)
+			L.apply_effect(4,INCINERATE)
+			L.adjustFireLoss(10)
+	qdel(src)
+
+/obj/item/weapon/sunlight_spear/proc/bang(var/turf/T , var/mob/living/carbon/M)					// Added a new proc called 'bang' that takes a location and a person to be banged.
+	if (locate(/obj/item/weapon/cloaking_device, M))			// Called during the loop that bangs people in lockers/containers and when banging
+		for(var/obj/item/weapon/cloaking_device/S in M)			// people in normal view.  Could theroetically be called during other explosions.
+			S.active = 0										// -- Polymorph
+			S.icon_state = "shield0"
+
+//Checking for protections
+	var/eye_safety = 0
+	var/ear_safety = 0
+	if(iscarbon(M))
+		eye_safety = M.eyecheck()
+		if(ishuman(M))
+			if(istype(M:l_ear, /obj/item/clothing/ears/earmuffs) || istype(M:r_ear, /obj/item/clothing/ears/earmuffs))
+				ear_safety += 2
+			if(HULK in M.mutations)
+				ear_safety += 1
+			if(istype(M:head, /obj/item/clothing/head/helmet))
+				ear_safety += 1
+
+//Flashing everyone
+	if(eye_safety < FLASH_PROTECTION_MODERATE)
+		flick("e_flash", M.flash)
+		M.Stun(1)
+		M.Weaken(5)
+			//Vaurca damage 15/01/16
+		var/mob/living/carbon/human/H = M
+		if(isvaurca(H))
+			var/obj/item/organ/eyes/E = H.get_eyes()
+			if(!E)
+				return
+			usr << span("alert", "Your eyes burn with the intense light of the flash!.")
+			E.damage += rand(10, 11)
+			if(E.damage > 12)
+				M.eye_blurry += rand(3,6)
+			if (E.damage >= E.min_broken_damage)
+				M.sdisabilities |= BLIND
+			else if (E.damage >= E.min_bruised_damage)
+				M.eye_blind = 5
+				M.eye_blurry = 5
+				M.disabilities |= NEARSIGHTED
+				spawn(100)

```

Comment:
Timer. There's already a proc defined on mob for clearing NEARSIGHTED you could use.